helix-term/src/ui/picker.rs

```diff
@@ -346,22 +365,56 @@ impl<T> Picker<T> {
         self.cursor = 0;
     }
 
-    pub fn move_up(&mut self) {
+    /// Move the cursor by a number of lines, `amount`.
+    ///
+    /// Positive numbers move the cursor down, negative numbers move it up. The cursor cycles
+    /// through the entries.
+    pub fn move_by(&mut self, amount: isize) {
+        if self.matches.is_empty() {
+            return;
+        }
+
+        let cursor: isize = self.cursor.try_into().unwrap();
+        let len: isize = self.matches.len().try_into().unwrap();
+
+        self.cursor = cursor
+            .saturating_add(amount)
+            .rem_euclid(len)
+            .try_into()
+            .unwrap();
+    }
+
+    /// Move the cursor down by exactly one page. After the last page comes the first page.
+    pub fn page_up(&mut self) {
+        self.move_by(-(self.completion_height as isize));
```

Comment:
Casting `usize` to `isize` and back isn't safe for all values. Use an enum to specify direction like we do for `fn scroll`: https://github.com/helix-editor/helix/blob/3a3f67115e4f8993ae20ff397cc9543c13bfbeb5/helix-term/src/commands.rs#L1250

Reply:
Then no `rem_euclid` is necessary and the code is more idiomatic without unwraps.

Reply:
> Casting `usize` to `isize` and back isn't safe for all values

Out of curiosity, which values will this be problematic for ?

Reply:
Any number between `isize::MAX` and `usize::MAX`. It's a very large number, but if there's no reason to cast back and forth between `isize` and `usize`, why not _shrug_

Reply:
This part should be safe because `completion_height` is `u16`. I think the lowest possible architecture size supported is 32 bits so means `u32` is the smallest, so it shouldn't be an issue. Correct me if I am wrong.

Reply:
Yeah true, but you could specify an `isize` that's larger than `u16`. `move_up(isize::MAX)` is wrong for example